File: test-suite-consul-graal/src/main/java/io/micronaut/consul/graal/HelloClient.java
package io.micronaut.consul.graal;

import io.micronaut.context.annotation.Requires;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.client.annotation.Client;

@Requires(property = "spec.name", value = "ConsulTest")
@Client(id = "service-discovery")
public interface HelloClient {

    @Get("/hello/{name}")
    String sayHi(String name);
}

File: test-suite-docs-python/src/test/python/micronaut/discovery/docs/HelloClient.py
# tag::imports[]
from abc import ABC, abstractmethod

from micronaut.http.annotation import Get
from micronaut.http.client.annotation import Client
# end::imports[]


# tag::class[]
@Client(id="hello-world")
class HelloClient(ABC):

    @Get("/hello")
    @abstractmethod
    def hello(self) -> str:
        ...
# end::class[]
